frb_codegen/src/generator/dart/mod.rs

```diff
@@ -467,53 +620,106 @@ fn generate_api_func(func: &IrFunc, ir_file: &IrFile) -> GeneratedApiFunc {
     }
 }
 
-fn generate_api2wire_func(ty: &IrType, ir_file: &IrFile, block_index: BlockIndex) -> String {
-    if let Some(body) = TypeDartGenerator::new(ty.clone(), ir_file, None).api2wire_body(block_index)
-    {
+fn rust_wire_to_dart_wire(ty: &str) -> Cow<str> {
+    let ty = ty.trim();
+    if matches!(ty, "void" | "WireSyncReturnStruct") {
+        return ty.into();
+    }
+    if ty.starts_with("*mut") {
```

Comment:
Well, seems a bit brittle... We are:

1. rely on string literals (void and WireSyncReturnStruct)
2. rely on type name prefixes (`*mut`)

In other words, we are playing with *strings*, while Rust has already given us such a rich type system. Maybe we can utilize Rust's type system?

Reply:
This is a stopgap for now since there isn't a cbindgen/ffigen pipeline for WASM. I'll see if I can clean this up.

Reply:
I see. Maybe we can:

1. `void` -> extract into a constant, since it is a keyword so constant is ok
2. WireSyncReturnStruct -> this is generated by our code, say in `our_code_1.rs`. Then in `our_code_1.rs` we create a constant instead of hardwiring this name, and use that const here as well
3. `*mut` -> maybe we can create a new function like `is_pointer` or something like that.